File: go/raindrops/raindrops.go
package raindrops

import (
	"strconv"
)

// Convert an int to a "raindrop"
// Just your basic FizzBuzz, really
func Convert(i int) string {
	var out = ""
	if i%3 == 0 {
		out +="Pling"
	}
	if i%5 == 0 {
		out += "Plang"
	}
	if i%7 == 0 {
		out += "Plong"
	}
	if out == "" {
		return strconv.Itoa(i)
	}
	return out
}

File: ocaml/raindrops/raindrops.ml
let raindrop i =
  match i mod 3, i mod 5, i mod 7 with
    0, 0, 0 -> "PlingPlangPlong"
  | 0, 0, _ -> "PlingPlang"
  | 0, _, 0 -> "PlingPlong"
  | _, 0, 0 -> "PlangPlong"
  | 0, _, _ -> "Pling"
  | _, 0, _ -> "Plang"
  | _, _, 0 -> "Plong"
  | _       -> string_of_int i
